test/test-BlueprintParser.cc

```diff
@@ -865,8 +884,8 @@ TEST_CASE("Report error when named type references itself in array", "[blueprint
     ParseResult<Blueprint> blueprint;
     SectionParserHelper<Blueprint, BlueprintParser>::parse(source, BlueprintSectionType, blueprint, ExportSourcemapOption, Models(), &blueprint);
 
-    REQUIRE(blueprint.report.error.code == MSONError);
-    SourceMapHelper::check(blueprint.report.error.location, 48, 26);
+    REQUIRE(blueprint.report.error.code == Error::OK);
+    REQUIRE(blueprint.report.warnings.empty());
```

Comment:
Here we need to add `REQUIRE`s to check that base type of `Comment` is `mson::ImplicitObjectBaseType`.